UpsertSubscriptionToMigrate/handler.ts

```diff
@@ -0,0 +1,268 @@
+        ON CONFLICT ("subscriptionId")
+        DO UPDATE
+            SET "organizationFiscalCode" = "excluded"."organizationFiscalCode",
+            "serviceVersion" = "excluded"."serviceVersion",
+            "serviceName" = "excluded"."serviceName"
+            WHERE "ServicesMigration"."Services"."status" <> '${excludeStatus}'
+            AND "ServicesMigration"."Services"."serviceVersion" < "excluded"."serviceVersion"
+    ` as NonEmptyString;
+
+export const log = (d: unknown): void => {
+  throw new Error(`To be implement ${d}`);
+};
+
+export const storeDocumentApimToDatabase = (
+  apimClient: ApiManagementClient,
+  config: IConfig,
+  pool: PoolClient,
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+) => (
+  retrievedDocument: RetrievedService
+): TE.TaskEither<DomainError, QueryResult | void> =>
+  pipe(
+    retrievedDocument.serviceId,
+    // given the subscription, retrieve it's apim object
+    id => getApimOwnerBySubscriptionId(config, apimClient, id),
+    TE.chain(apimSubscription =>
+      pipe(
+        // given the subscription apim object, retrieve its owner's detail
+        getApimUserBySubscription(config, apimClient, apimSubscription),
+        // We only consider subscription owned by a Delegate,
+        //   otherwise we just ignore the document
+        // This because migration are meant to work only from a Delegate to its Organization,
+        //   not to migrate subscriptions between organizations
+        TE.chain(apimUser =>
+          ApimDelegateUserResponse.is(apimUser)
+            ? // continue processing incoming document
+              pipe(
+                { apimSubscription, apimUser },
+                apimData => mapDataToTableRow(retrievedDocument, apimData),
+                createUpsertSql(config),
+                sql => queryDataTable(pool, sql),
+                res => {
+                  trackProcessedServiceDocument(telemetryClient)(
+                    retrievedDocument
+                  );
+                  return res;
+                }
+              )
+            : // processing is successful, just ignore the document
+              TE.of<DomainError, QueryResult | void>(
+                trackIgnoredIncomingDocument(telemetryClient)(retrievedDocument)
+              )
+        )
+      )
+    )
+  );
+
+export const createHandler = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  client: Pool,
+  telemetryClient: ReturnType<typeof initTelemetryClient>
+): Parameters<typeof withJsonInput>[0] =>
+  withJsonInput(
+    async (context, item): Promise<void> => {
+      const logPrefix = context.executionContext.functionName;
+      const pool = await client.connect();
+      return pipe(
+        item,
+        IncomingQueueItem.decode,
+        E.mapLeft(failures => {
```

Comment:
I think we should extract this kind of functions since we use them also in other places with slight changes.
I think we miss an error handling common library to do this things.